### ISSUE TITLE
CA1849 should suggest using `ServiceProviderServiceExtensions.CreateAsyncScope`

### ISSUE THREAD

**Issue by meziantou** [dotnet/roslyn-analyzers#6271]
### Analyzer

**Diagnostic ID**: [CA1849](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1849) Call async methods when in an async method

### Describe the improvement

CA1849 should report [ServiceProviderServiceExtensions.CreateScope](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.createscope) and suggest using the async equivalent [ServiceProviderServiceExtensions.CreateAsyncScope](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serviceproviderserviceextensions.createasyncscope) when available.

````
// <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.0.0" />

using Microsoft.Extensions.DependencyInjection;

IServiceProvider serviceProvider = null;
using var s = serviceProvider.CreateScope(); // report diagnostic
await using var scope = serviceProvider.CreateAsyncScope(); // ok
````

The `CreateAsyncScope` is not async itself, but return an `IAsyncDisposable` instead of `IDisposable`. So, it will allow calling `DisposeAsync()` instead of `Dispose()`.
